python/amici/petab_objective.py

```diff
@@ -133,72 +124,50 @@ def simulate_petab(
     sllh = aggregate_sllh(amici_model=amici_model, rdatas=rdatas,
                           parameter_mapping=parameter_mapping)
 
-    # TODO: implement me
-    # # Compute total fim
-    # fim = None
-    # # Compute total s2llh
-    # s2llh = None
-    # # Compute total res
-    # res = None
-    # # Compute total sres
-    # sres = None
-
-    # log results
+    # Log results
     sim_cond = petab_problem.get_simulation_conditions_from_measurement_df()
-
     for i, rdata in enumerate(rdatas):
         logger.debug(f"Condition: {sim_cond.iloc[i, :].values}, status: "
                      f"{rdata['status']}, llh: {rdata['llh']}")
 
     return {
         LLH: llh,
         SLLH: sllh,
-        # FIM: fim,
-        # S2LLH: s2llh,
-        # RES: res,
-        # SRES: sres,
         RDATAS: rdatas
     }
 
 
-def edatas_from_petab(
-        model: amici.Model,
+def create_parameterized_edatas(
+        amici_model: amici.Model,
         petab_problem: petab.Problem,
         problem_parameters: Dict[str, numbers.Number],
+        scaled_parameters: bool = False,
+        parameter_mapping: List[petab.ParMappingDictQuadruple] = None,
         simulation_conditions: Union[pd.DataFrame, Dict] = None,
-        parameter_mapping: List[petab.ParMappingDictTuple] = None,
-        scaled_parameters: Optional[bool] = False
 ) -> List[amici.ExpData]:
-    """
-    Create list of :class:`amici.amici.ExpData` objects for PEtab problem.
-
-    Sets timepoints, fixed parameters (including preequilibration),
-    non-fixed parameters, and observed data and sigmas.
-
-    :param model:
-        AMICI model.
+    """Create list of :class:amici.ExpData objects with parameters filled in.
 
+    :param amici_model:
+        AMICI Model assumed to be compatible with ``petab_problem``.
     :param petab_problem:
-        PEtab problem
-
+        PEtab problem to work on.
     :param problem_parameters:
-        Dictionary mapping parameter names of the PEtab problem to
-        parameter values
-
-    :param simulation_conditions:
-        Result of petab.get_simulation_conditions. Can be provided to save
-        time if this has be obtained before.
-
-    :param parameter_mapping:
-        Optional precomputed PEtab parameter mapping for efficiency.
-
+        Run simulation with these parameters. If None, PEtab `nominalValues`
+        will be used). To be provided as dict,  mapping PEtab problem
+        parameters to SBML IDs.
     :param scaled_parameters:
         If True, problem_parameters are assumed to be on the scale provided
-        in the PEtab parameter table and will be unscaled. If False,
-        they are assumed to be in linear scale.
+        in the PEtab parameter table and will be unscaled. If False, they
+        are assumed to be in linear scale.
+    :param parameter_mapping:
+        Optional precomputed PEtab parameter mapping for efficiency.
```

Comment:
Has to be from create_parameter_mapping, right?

Reply:
yes, will document.